src/EventTypes.ts

```diff
@@ -0,0 +1,62 @@
+export type GraphEvents = Partial<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | Record<UnsupportedTypeNetworkEvents, (params?: any) => void>
+  | Record<'click', (params: SingleClickData) => void>
+  | Record<
+      | 'doubleClick'
+      | 'oncontext'
+      | 'hold'
+      | 'release'
+      | 'select'
+      | 'selectNode'
+      | 'selectEdge'
+      | 'dragStart'
+      | 'dragging'
+      | 'dragEnd',
+      (params: BaseClickData) => void
+    >
+>;
+
+type UnsupportedTypeNetworkEvents =
+  | 'deselectNode'
+  | 'deselectEdge'
+  | 'controlNodeDragging'
+  | 'controlNodeDragEnd'
+  | 'hoverNode'
+  | 'blurNode'
+  | 'hoverEdge'
+  | 'blurEdge'
+  | 'zoom'
+  | 'showPopup'
+  | 'hidePopup'
+  | 'startStabilizing'
+  | 'stabilizationProgress'
+  | 'stabilizationIterationsDone'
+  | 'stabilized'
+  | 'resize'
+  | 'initRedraw'
+  | 'beforeDrawing'
+  | 'afterDrawing'
+  | 'animationFinished'
+  | 'configChange';
+
+export interface BaseClickData {
+  nodes: string[];
+  edges: string[];
+  event: unknown; // TODO add event type
```

Comment:
what about that? Why this is unknown also?

Reply:
this should be MouseEvent or its react equivalent probably

Reply:
I have trouble finding some existing type, maybe it is a custom from vis js?

Reply:
We should try and add the properties that exist... 
```
type SubEvent= {
    "pointers": unknown[],
    "changedPointers": PointerEvent[
    ],
    "pointerType": string,// or a union probably?
    "srcEvent": PointerEvent,
    "isFirst": boolean,
    "isFinal": boolean,
    "eventType": number,// enum?
    "center": {
        "x": number,
        "y": number
    },
    "timeStamp": number,
    "deltaTime": number,
    "angle": number,
    "distance": number,
    "deltaX": number,
    "deltaY": number,
    "offsetDirection": number,
    "overallVelocityX": number,
    "overallVelocityY": number,
    "overallVelocity": number,
    "scale": number,
    "rotation": number,
    "maxPointers": number,
    "velocity": number,
    "velocityX": number,
    "velocityY": number,
    "direction": number,
    "target": HTMLCanvasElement,
    "type": "doubletap",
    "tapCount": number
}
```

Reply:
Lets do it separately, I will include this in the issue